nemo/collections/tts/modules/fastpitch.py

```diff
@@ -222,6 +273,9 @@ def forward(
         attn_prior=None,
         mel_lens=None,
         input_lens=None,
+        gst_ref_spec=None,
+        gst_ref_spec_lens=None,
```

Comment:
I would suggest the function take the output of the reference encoder/GST as input, and move the logic of running the GST inference to the model (or elsewhere). At training time a random reference can be used, but at inference it should be possible to provide a centroid or interpolated embedding computed across several references from a speaker.

For this specific implementation, it looks like you might be able to just preprocess it and pass in the final summed "speaker_embedding"